components/entitlement/org.wso2.carbon.identity.entitlement/src/test/java/org/wso2/carbon/identity/entitlement/persistence/PolicyPersistenceManagerTest.java

```diff
@@ -0,0 +1,305 @@
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy2, true);
+
+        policyPersistenceManager.addPolicy(samplePDPPolicy1);
+        policyPersistenceManager.addPolicy(samplePDPPolicy2);
+
+        // Verify the policies that are not active.
+        assertNull(policyPersistenceManager.getReferencedPolicy(samplePDPPolicy1.getPolicyId()));
+        assertEquals(policyPersistenceManager.getReferencedPolicy(samplePDPPolicy2.getPolicyId()),
+                samplePDPPolicy2.getPolicy());
+    }
+
+    @Test(priority = 10)
+    public void testGetPolicyOrder() throws Exception {
+
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy1, true);
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy3, true);
+
+        policyPersistenceManager.addPolicy(samplePDPPolicy1);
+        policyPersistenceManager.addPolicy(samplePDPPolicy3);
+
+        // Verify the policy order.
+        assertEquals(policyPersistenceManager.getPolicyOrder(samplePDPPolicy1.getPolicyId()), 0);
+        assertEquals(policyPersistenceManager.getPolicyOrder(samplePDPPolicy3.getPolicyId()),
+                samplePDPPolicy3.getPolicyOrder());
+    }
+
+    @Test(priority = 11)
+    public void testListPDPPolicy() throws Exception {
+
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy1, true);
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy2, true);
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy3, true);
+
+        policyPersistenceManager.addPolicy(samplePDPPolicy1);
+        policyPersistenceManager.addPolicy(samplePDPPolicy2);
+        policyPersistenceManager.addPolicy(samplePDPPolicy3);
+
+        // Verify the number of published policies.
+        List<String> policyIds = policyPersistenceManager.listPublishedPolicyIds();
+        assertEquals(policyIds.size(), 3);
+
+        // Verify the number of ordered policy identifiers.
+        String[] orderedPolicyIdentifiers = policyPersistenceManager.getOrderedPolicyIdentifiers();
+        assertEquals(orderedPolicyIdentifiers.length, 3);
+
+        // Verify the number of active policies.
+        String[] activePolicies = policyPersistenceManager.getActivePolicies();
+        assertEquals(activePolicies.length, 2);
+    }
+
+    @Test(priority = 12)
+    public void testUpdatePDPPolicy() throws Exception {
+
+        policyPersistenceManager.addOrUpdatePolicy(samplePAPPolicy1, true);
+        policyPersistenceManager.addPolicy(samplePDPPolicy1);
+
+        // Update Policy order.
+        policyPersistenceManager.updatePolicy(orderedSamplePDPPolicy1);
+        PolicyStoreDTO orderUpdatedPDPPolicy =
+                policyPersistenceManager.getPublishedPolicy(samplePDPPolicy1.getPolicyId());
+        assertEquals(policyPersistenceManager.getPolicyOrder(samplePDPPolicy1.getPolicyId()),
+                orderedSamplePDPPolicy1.getPolicyOrder());
+        assertTrue(orderUpdatedPDPPolicy.isActive());
```

Comment:
Minor change.. Better to initialize `orderUpdatedPDPPolicy` just before the assertTrue() as follows.

```java
       assertEquals(policyPersistenceManager.getPolicyOrder(samplePDPPolicy1.getPolicyId()),
                orderedSamplePDPPolicy1.getPolicyOrder());
          PolicyStoreDTO orderUpdatedPDPPolicy =
                policyPersistenceManager.getPublishedPolicy(samplePDPPolicy1.getPolicyId());
      assertTrue(orderUpdatedPDPPolicy.isActive());
```

Reply:
https://github.com/wso2/carbon-identity-framework/compare/2f503927491810d4d2cc8b7ea09b65f587525e2e..585e1901da74cee150023ab7584efca46ed666d1